model/src/main/kotlin/Identifier.kt

```diff
@@ -126,8 +126,22 @@ data class Identifier(
         sanitizedProperties.joinToString(separator) { it.encodeOr(emptyValue) }
 }
 
+/**
+ * A map of lowercase aliases for types that were renamed.
+ */
+private val typeAliases = mapOf(
```

Comment:
I'm not sure whether a map is the right data structure here. To me, aliases are a set of alternative names, not a one-to-one relationship. I'd propose more something like we have for `VcsType` aliases, i.e. a list of names that are all treated equally (when consuming), and by convention use the first entry in the list when producing.

Reply:
> that is not possible because the matching logic is inside the `model` module which is a dependency of the package manager plugins.

Well, the matching logic could stay in the `model`, but the definition of aliases could still be moved to the plugins, simply by making `projectType` a list of aliases instead of a single string, as described above.

Reply:
How should the logic in `model` get access to the aliases if they are defined in Gradle modules that depend on `model`? It doesn't even know what `projectType`s the plugins define.